lib/tasks/eu_exit_business_finder.rake

```diff
@@ -0,0 +1,108 @@
+namespace :eu_exit_business_finder do
+  task update_subscriber_list_titles_and_facet_values: [:environment] do
+    subscriber_lists = SubscriberList
+      .where("links::text != '{}'")
+      .where("ARRAY(SELECT json_object_keys(links)) <@ Array[:keys]", keys: [:facet_values])
+    update_facet_values(subscriber_lists)
+  end
+
+  # Iterate through the affected subscriber lists and if we encounter any
+  # facet values that need replacing with a newer one, we insert the new one
+  # to the array (inserted 'in place' to retain original order). Then remove
+  # the old facet values in bulk and write only the unique values that remain.
+  def update_facet_values(subscriber_lists)
+    subscriber_lists.each do |list|
+      links = list.links[:facet_values][:any]
+      updated_links = links.map { |original_val| replace_facet_val(original_val) || original_val }
+      updated_links = (updated_links - EuExitFacetMigrationConfig::facet_values_to_remove).uniq
+      list.write_attribute('links', facet_values: { any: updated_links })
+      list.write_attribute('title', title_from_facets(updated_links))
+      list.save
+    end
+  end
+
+  def replace_facet_val(original_facet_val)
+    EuExitFacetMigrationConfig::facet_values_to_replace
+      .select { |replace_val| replace_val[:old_facet_value] == original_facet_val }
+      .map { |replace_val| replace_val[:new_facet_value] }[0]
+  end
+
+  # Title is constructed from finder content item, so we have to retrieve the
+  # facet titles from the JSON using their content IDs.
+  def title_from_facets(links)
```

Comment:
I assume that this method is being passed a list of the content-ids of facet-values, and that the list is in correct order? i.e. sectors, then org activity etc.

If that's the case, then this method should be very simple. 

You what the subscription title prefix is, so it should just be a case of looping through the links array, getting the title of the content item of the facet value from publishing-api, and concatenating that to the "subscription title prefix"

Also, I think "links" should be renamed to "facet_values" so we're clear what they are.

e.g.
```
subscription_title_prefix = "EU Exit guidance for your business in the following "
# category or categories code goes here

titles = facet_values.map do |facet_value|
  if EuExitFacetMigrationConfig.facet_value_labels_to_replace[facet_value].present?
    return "'#{EuExitFacetMigrationConfig.facet_value_labels_to_replace[facet_value]}'"
 end
  content_item = GdsApi.publishing_api_v2.get_content(facet_value).to_h
  "'#{content_item['title']}'"
end
```
subscription_title_prefix + titles.join(', ')

Reply:
Definitely more readable 👍think making one request to Publishing API (for business finder content item) is more efficient than lots of little ones, but not an issue at this scale.

Have refactored now.